component/namespace-policies.jsonnet

```
@@ -183,11 +183,71 @@ local disallowReservedNamespaces = kyverno.ClusterPolicy('disallow-reserved-name
   },
 };
 
+/**
+  * Disallow auxiliary labels and annottions
+  * This policy will:
+  * - Check annotations and labels against a whitelist.
+  * - Deny namespace creation or modification.
+  */
+local validateNamespaceMetadata = kyverno.ClusterPolicy('validate-namespace-metadata') {
+  local validateObject = function(key, whitelist) {
+    name: 'validate-%s' % key,
+    match: common.MatchNamespaces(),
+    exclude: common.BypassNamespaceRestrictionsSubjects(),
+    preconditions: {
+      all: [
+        {
+          key: '{{request.operation}}',
+          operator: 'In',
+          value: [ 'CREATE', 'UPDATE' ],
+        },
+      ],
+    },
+    validate: {
+      message: 'The following %s are allowed: %s' % [ key, std.join(', ', whitelist) ],
+      foreach: [
+        {
+          list: 'request.object.metadata.%s | map(&{key: @}, keys(@))' % key,
+          deny: {
+            conditions: {
+              all: [
+                {
+                  key: '{{element.key}}',
+                  operator: 'NotEquals',
+                  value: w,
+                }
+                for w in whitelist
+              ],
+            },
+          },
+        },
+      ],
+    },
+  },
+  metadata+: {
+    annotations+: {
+      // Workaround for bug in `kyverno-cli test` command.
+      // Commodore generates an empty annotation object (`annotations: {}`)
+      // which trips up Kyverno. A non empty object or `annotations: null` is valid.
+      // Ensure a label to not have an empty object.
+      'policies.kyverno.io/category': 'Namespace Management',
+    },
```

Besides this, why not also also add some common labels (from https://syn.tools/syn/references/labels-guide.html) ?
```suggestion
    annotations+: {
      // Workaround for bug in `kyverno-cli test` command.
      // Commodore generates an empty annotation object (`annotations: {}`)
      // which trips up Kyverno. A non empty object or `annotations: null` is valid.
      // This ensures that `annotations` key does not have an empty object.
      'policies.kyverno.io/category': 'Namespace Management',
    },
```

---

Common labels are added further down in the file.